types/underscore/underscore-tests.ts

```diff
@@ -1,9 +1,157 @@
+/**************************************
+ * Common Testing Types and Variables *
+ **************************************/
 declare const $: any;
 declare const window: any;
 declare const alert: (msg: string) => any;
 declare const console: {log: any};
+
+declare const context: object;
+
+// string record (also used to represent a generic record)
+interface StringRecord {
+    a: string;
+    b: string;
+}
+
+interface AugmentedList extends _.List<StringRecord> {
+    notAListProperty: boolean;
+}
+
+interface ExplicitDictionary extends _.Dictionary<StringRecord> {
+    a: StringRecord;
+    b: StringRecord;
+    c: StringRecord;
+}
+
+declare const shallowProperty: 'a';
+declare const deepProperty: ['a', 'length'];
```

Comment:
As a point of interest, as part of doing this I found that there was an issue similar to #46650 in the `Iteratee` and `IterateeResult` types that I plan to fix in a quick PR after this merges.

Reply:
I ended up having to give up on this for the moment due to errors that occur in TS3.0 but not TS3.1, so this will probably have to be included in future TS minimum version upgrade work.

As another point of interest here, I found that the `isArray` function doesn't work well with type unions that contain a readonly array or tuple, and super sadly I also could not find a solution there that would work with both a readonly array and an `any`.